pkg/extensions/management.go

```diff
@@ -0,0 +1,25 @@
+package extensions
+
+import (
+	jenkinsv1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	jenkinsclientv1 "github.com/jenkins-x/jx/pkg/client/clientset/versioned/typed/jenkins.io/v1"
+	"github.com/jenkins-x/jx/pkg/log"
+)
+
+func GetChildrenRecursively(ext jenkinsv1.Extension, configLookup map[string]jenkinsv1.ExtensionConfig, lookupByUUID map[string]jenkinsv1.Extension, extensionsClient jenkinsclientv1.ExtensionInterface, ns string, verbose bool) ([]jenkinsv1.Extension, error) {
+	result := make([]jenkinsv1.Extension, 0)
```

Comment:
I would suggest that you make `result` a set (well, a `map`) and make this method return as set (map) to return unique children, rather than making the caller do the de-duplication.

Reply:
Good idea.